include/pybind11/stl/filesystem.h

```diff
@@ -13,21 +13,30 @@
 #include <string>
 
 #ifdef __has_include
-#    if defined(PYBIND11_CPP17) && __has_include(<filesystem>)
-#        include <filesystem>
-#        define PYBIND11_HAS_FILESYSTEM 1
+#    if defined(PYBIND11_CPP17)
+#        if __has_include(<filesystem>) && \
+          PY_VERSION_HEX >= 0x03060000
+#            include <filesystem>
+#            define PYBIND11_HAS_FILESYSTEM 1
+namespace pybind11_filesystem_alias = std::filesystem;
+#        elif __has_include(<experimental/filesystem>)
+#            include <experimental/filesystem>
+#            define PYBIND11_HAS_EXPERIMENTAL_FILESYSTEM 1
+namespace pybind11_filesystem_alias = std::experimental::filesystem;
```

Comment:
Actually, any reason it needs to / should be in the global namespace? Do we want people downstream to reuse it? Even if we do wouldn't be better to just make it part of the pybind11 namespace?

Reply:
Thoughts @rwgk ?

Reply:
I agree, totally doesn't need to be global

Reply:
Yeah, I would try sticking it in detail and see if anything blows up.

Reply:
On second glance, I can't actually think of any way to completely remove this from the public API here. I can put it under the pybind11 namespace if that makes us feel better. Between `pybind11_filesystem_alias` or `pybind11::filesystem_alias`, I think the intent is clear enough. They're both unlikely to conflict with any user namespaces, they both have clear intent, and even if you were to use it, the can't cause any real damage; it's just a namespace alias.

Reply:
@Llcoolsouder the `detail` namespace is how we denote private APIs / functions in pybind11. Putting under the pybind11 namespace would still be better though.